modules/compiler/include/compiler/module.h

```diff
@@ -339,11 +339,117 @@ struct KernelInfo {
   std::uint64_t private_mem_size;
 
   /// @brief Values of reqd_work_group_size attribute if it exists.
-  cargo::optional<std::array<size_t, 3>> work_group;
+  cargo::optional<std::array<size_t, 3>> reqd_work_group_size;
+
+  std::size_t getNumArguments() const { return argument_types.size(); }
+
+  /// @brief Returns the reqd_work_group_size if present, else an all-zeros
+  /// array.
+  std::array<size_t, 3> getReqdWGSizeOrZero() const {
+    return reqd_work_group_size.value_or(std::array<size_t, 3>{0, 0, 0});
+  }
 };  // class KernelInfo
 
-/// @brief Kernel info callback type.
-using KernelInfoCallback = std::function<void(KernelInfo)>;
+/// @brief Class for managing program information.
```

Comment:
I think this could say a little more, what kind of information. it seems to be a container of multiple kernel informations (and not much else).

Reply:
good idea. yeah that's basically it. I stole it from the CL side of things, which doesn't need anything more than that.